drivers/misc/rwbuffer.c

```diff
@@ -1276,7 +1276,10 @@ int rwb_flush(FAR struct rwbuffer_s *rwb)
 {
   int ret;
 
+#if CONFIG_DRVR_WRDELAY != 0
   ret = rwb_forcetake(&rwb->wrsem);
```

Comment:
not match with 1285

Reply:
Maybe better to apply next change:
```
#if defined(CONFIG_DRVR_WRITEBUFFER) && CONFIG_DRVR_WRDELAY != 0
static int rwb_forcetake(FAR sem_t *sem)
{
  int result;
  int ret = OK;

  do
    {
      result = rwb_semtake(sem);

      /* The only expected failure is if the thread is canceled */

      DEBUGASSERT(result == OK || result == -ECANCELED);
      if (ret == OK && result < 0)
        {
          /* Remember the first error */

          ret = result;
        }
    }
  while (result < 0);

  return ret;
}
#else
#  define rwb_flush(s)
#endif
```
and
```
#if defined(CONFIG_DRVR_WRITEBUFFER) && CONFIG_DRVR_WRDELAY != 0
#  define rwb_semgive(s) nxsem_post(s)
#else
#  define rwb_semgive(s)
#endif
```
?

Reply:
Also maybe block
```
#ifdef CONFIG_DRVR_WRITEBUFFER
  sem_t         wrsem;           /* Enforces exclusive access to the write buffer */
  struct work_s work;            /* Delayed work to flush buffer after a delay with no activity */
  uint8_t      *wrbuffer;        /* Allocated write buffer */
  uint16_t      wrnblocks;       /* Number of blocks in write buffer */
  off_t         wrblockstart;    /* First block in write buffer */
#endif
```
in `struct rwbuffer_s` can be made conditional on `CONFIG_DRVR_WRDELAY != 0`?